### PR TITLE
Autoscale Cleanup

### PR DESCRIPTION
**Pull request by jfryman**
This PR contains several final cleanup items necessary for publication to the greater world. This includes:

* Cleanup of Mistral Workflow
* Addition of Governor Code
* 0.8 Mistral Support
* Real ChatOps Support

### CODE REVIEW

File: packs/autoscale/actions/workflows/asg_deflate.yaml [hunk 1]
```diff
@@ -16,83 +16,82 @@ workflows:
...
         on-success:
           - get_deflate_delay
       get_deflate_delay:
         action: st2.kv.get
         input:
-          key: 'asg.{$.asg}.deflate_delay'
+          key: 'asg.<% $.asg %>.deflate_delay'
         publish:
-          deflate_delay: int($.get_deflate_delay.result)
+          deflate_delay: <% int($.get_deflate_delay.result) %>
```
Comment (dzimine) [inline packs/autoscale/actions/workflows/asg_deflate.yaml]: I wonder if int() is still needed here...

File: packs/autoscale/actions/workflows/asg_deflate.yaml [hunk 2]
```diff
@@ -16,83 +16,82 @@ workflows:
...
+              ((<% $.last_deflate_timestamp %> + (<% $.deflate_delay %> * 60)) < <% $.current_epoch %>)
         on-error:
           - chatops_too_soon
       chatops_delete_nodes:
         action: slack.post_message
         input:
-          message: "ASG {$.asg} is a little bloated, it seems. Going to trim by {$.deflate_by} nodes..."
-          channel: '{$.channel}'
+          message: "ASG is a little bloated, it seems. Going to trim by <% $.deflate_by %> nodes..."
+          channel: '<% $.channel %>'
```
Comment (dzimine) [inline packs/autoscale/actions/workflows/asg_deflate.yaml]: 1) check if @Kami fixed slack to take channel from config
2) check if ' ' are needed, YAQL now supposed to preserve types.

File: packs/autoscale/actions/workflows/asg_delete.yaml [hunk 3]
```diff
@@ -15,116 +15,134 @@ workflows:
...
           - delete_last_deflate_timestamp
       delete_deflate_by_key:
         action: st2.kv.delete
         input:
-          key: 'asg.{$.name}.deflate_by'
+          key: 'asg.<% $.name %>.deflate_by'
         on-success:
           - delete_deflate_delay_key
       delete_deflate_delay_key:
         action: st2.kv.delete
```
Comment (dzimine) [inline packs/autoscale/actions/workflows/asg_delete.yaml]: consider one-liner: `st2.kv.delete key="'asg.<% $.name %>.deflate_delay'"`

**Comment by dzimine** [comment]
It may be the most advanced Mistral workflow so far :) 
Good stuff. Some soft comments inside.

File: packs/autoscale/actions/workflows/asg_deflate.yaml (hunk 1 above)
Comment (jfryman) [inline packs/autoscale/actions/workflows/asg_deflate.yaml]: This I need to test. I recall running into an issue with the K/V store and types, which is why the cast is here. This is certainly one thing that needs testing tomorrow. 

-James

File: packs/autoscale/actions/workflows/asg_delete.yaml (hunk 3 above)
Comment (jfryman) [inline packs/autoscale/actions/workflows/asg_delete.yaml]: Ah! Good call. I was wondering how to make all the key writes a bit cleaner. Also have some thoughts on how to make multiple calls as well

-James

File: packs/autoscale/actions/workflows/asg_deflate.yaml (hunk 2 above)
Comment (jfryman) [inline packs/autoscale/actions/workflows/asg_deflate.yaml]: Channel is a user configurable value on a per-asg basis, and that's by design. Would be good to know if I can call config from workflows as a general rule though. 

-James

**Comment by jfryman** [comment]
This PR has gotten unwieldy!

**Pull request merged**